lsp/nls/src/linearization/building.rs

```diff
@@ -162,10 +165,48 @@ impl<'b> Building<'b> {
                     .as_option()
                     .and_then(|value_index| self.get_item_kind(current_file, value_index))
             }
-            // if declaration is a let binding resolve its value
-            TermKind::Declaration(_, _, ValueState::Known(value)) => {
-                self.get_item_kind(current_file, *value)
+            TermKind::Declaration {
+                value: ValueState::Known(value),
+                path: Some(idents),
+                ..
+            } => {
+                let item = self.get_item_kind(current_file, *value)?;
+                let item = self.resolve_reference(current_file, item)?;
+
+                let mut ids = idents.clone();
+                // (previous value, current value)
+                let (mut prev_item, mut curr_item) = (item, item);
+                while let Some(id) = ids.pop() {
+                    match curr_item {
+                        TermKind::Record(ref fields) => {
+                            let item = fields.get(&id)?;
+                            let item = self.get_item_kind(current_file, *item)?;
```

Comment:
Nitpick, but you're calling two things with different types with the same name (which is ok, sometimes, granted).

```suggestion
                            let item_kind = self.get_item_kind(current_file, *item)?;
```